File: typescript/packages/bench/src/exporters/jsonExporter.ts
/** JSON rendering of the row objects. */

import type { ReportRow } from "../contracts/reportContract";
import { AbstractExporter } from "../support/abstractExporter";

/** JSON rendering of the row objects. */
export class JsonExporter extends AbstractExporter {
    constructor() {
        super("json", "application/json");
    }

    override export(rows: readonly ReportRow[]): string {
        return JSON.stringify(rows);
    }
}

File: python/atelier_bench/exporters/json_exporter.py
"""json exporter."""

from __future__ import annotations

import json
from collections.abc import Sequence

from ..contracts.report_contract import ReportRow
from ..support.abstract_exporter import AbstractExporter


class JsonExporter(AbstractExporter):
    def __init__(self) -> None:
        super().__init__("json", "application/json")

    def export(self, rows: Sequence[ReportRow]) -> str:
        return json.dumps([row._asdict() for row in rows], separators=(",", ":"))
